gcc-11.3.0/gcc/gcc.c

```diff
@@ -8233,10 +8236,10 @@ get_sha1_gitoid (char **gitoid, FILE *command)
 	{
 	  ++line_cnt;
 	  if (line_cnt == 2)
-	    strcpy (line1, line);
+	    memcpy (line1, line, 1000);
```

Comment:
could you also add a comment why you read 1000 bytes (may be when defining line1 and line2)?